### PR TITLE
Add commentary to 9.3 Working with personal names

### PR DESCRIPTION
**Pull request by r12a**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/pull/72.html" title="Last updated on Jun 15, 2022, 2:46 PM UTC (dc58b3b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/72/0fc8fcc...dc58b3b.html" title="Last updated on Jun 15, 2022, 2:46 PM UTC (dc58b3b)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2816,7 +2816,9 @@ <h3>Working with time</h3>
 <section id="loc_names" class="subtopic">
 <h3>Working with personal names</h3>
 
- 
+
+<p>People who create web forms, databases, or ontologies are often unaware how different people’s names can be in other countries. They build their forms or databases in a way that assumes too much on the part of foreign users. This section provides guidelines for working with personal names from around the world.</p>
```
Comment (aphillips) [inline index.html]: It's probably just me, but I found the first sentence difficult to read--the false correspondence between `people` and `people` and the fact that `different` isn't an adjective. I also found `forms, databases, or ontologies` perhaps too restrictive? Perhaps:

> People who are creating applications that use personal names (in forms, databases, ontologies, and so forth),  are often unaware of how different personal names can be in other countries.

File: index.html [hunk 2]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
 </aside>
 
 
+<section id="loc_names_field_length">
+<h4>Field length &amp; composition</h4>
+
 	<div class="req" id="loc_names_given_family">
-	<p class="advisement">Check whether you really need to store or access given name and family name separately.</p>
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<p class="advisement">Check whether you really need to store or access given name(s) and family name(s) separately.</p>
```
Comment (aphillips) [inline index.html:2833]: perhaps leave off the parens? I assume you're trying to emphasize that there can be more than one of each, but perhaps using the plural is enough to do that.

File: index.html [hunk 3]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#singlefield">To split or not to split?</a> in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p>Names around the world differ greatly in composition and the order of components (see <a href="https://www.w3.org/International/questions/qa-personal-names">Personal names around the world</a>). This can create difficulties when you try to split a person's name into smaller parts for storage in a database and then retrieve them. Difficulties include understanding which part of a person's name belongs in which database field (especially when there are more or fewer parts than fields in the database), and dealing with ordering of name parts when retrieving someone's name from the database for actual use.</p>
```
Comment (aphillips) [inline index.html]: perhaps `and then retrieve them` => `and then reconstruct the name later`?

File: index.html [hunk 4]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
-	<div class="req" id="loc_names_first_last">
-	<p class="advisement">Try to avoid using the labels 'first name' and 'last name' in non-localized contexts.</p>
-	<details class="links"><summary>explanations &amp; examples</summary>
+<p>Bear in mind that names in some cultures can be quite a lot longer than your own. Make  fields long enough to enter long names. Also do not assume that a name will have more than one letter. In particular, avoid counting length in bytes (see [[#char_string]]) – do not assume that a four-character Japanese name in UTF-8 will fit in four bytes; you are likely to actually need 12.</p>
```
Comment (aphillips) [inline index.html]: I would probably paragraph break before `In particular`

Perhaps the discussion of length limits should link to section 5.5

File: index.html [hunk 5]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
+
+<section id="loc_names_segments">
+<h4>Guidelines for segmenting names</h4>
+
+
+<p>The guidelines in this section apply where a decision has been made that it is necessary to split up a person's name for storage or presentation.</p>
+
+
+<div class="req" id="loc_names_first_last">
+	<p class="advisement">Try to avoid using the labels 'first name' and 'last name' in non-localized contexts. Consider an alternative such as 'given name(s)' and 'family name(s)'.</p>
```
Comment (aphillips) [inline index.html]: I understand what you're doing here, but maybe more explanation is needed? The phrase "non-localized contexts" is somewhat mysterious. Perhaps add an example that is more direct, such as:

> For example, users in the United States expect forms to be labeled 'first name' and 'last name', but the generic, unlocalized, software (even in English) should use the non-positional labels 'given name' and 'family name' so that users in other countries can understand which value goes into which field. The form can then be localized into US English with "positional" labels for that audience.

File: index.html [hunk 6]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#splitting">Strategies for splitting up names</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
-	<div class="req" id="loc_names_extra_fields">
+<p>Use of the terms 'first' and 'last' can be confusing for people who normally write their family name followed by given names. However, bear in mind also that in some cultures this is still problematic (for example Icelanders, who don't actually have family names). However, short of very localized customization, this is probably the best we can do for a generic solution.</p>
```
Comment (aphillips) [inline index.html]: This kind of addresses what I wrote above. 

The Iceland example may be a bit confusing for casual readers (perhaps mention that the patronym isn't actually a family name)

File: index.html [hunk 7]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
-	<details class="links"><summary>explanations &amp; examples</summary>
+<p>For example, in some cases you may want to identify parts of a name so that you can sort a list of names alphabetically, or address them when contacting them, etc.</p>
+   
+<p>This extra field would also be useful for finding the appropriate name from a long list of name components, and for handling nicknames (which, for example, are commonly used to refer to people in Thailand).</p>
+<p>Sometimes you may opt for separate fields because you want to be able to use part of the name to address the person directly, or refer to them. For example, when Google+ refers to "David's contacts". Or perhaps it's because you want to send them emails with their name at the top. Note that not only may you have problems due to name syntax here, but you also have to account for varying expectations around the world with regards to formality (not everyone is happy for a stranger to call them by their given name). It may be better to ask separately, when setting up a profile for example, how that person would like you to address them.</p>
```
Comment (aphillips) [inline index.html]: Perhaps don't mention `Google+`, since it is defunct.

File: index.html [hunk 8]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#splitting">Strategies for splitting up names</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
-	<div class="req" id="loc_names_algorithms">
+<p>For example, don't assume that the order they will provide names in will be given followed by family.</p>
```
Comment (aphillips) [inline index.html]: This sentence is super hard to parse. Perhaps change `will provide` to `provide`?

File: index.html [hunk 9]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#splitting">Strategies for splitting up names</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
-	</div>
+    <details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
+</div>
+
+<p>For example, the v-card and h-card approach of implied “n” optimization could have difficulties with, say, Chinese names. The input form should be as clear as possible when telling people how to specify their name, so that you capture the data you think you need.</p>
```
Comment (aphillips) [inline index.html:2931]: `v-card`? `h-card`? `implied "n" optimization`? Something missing here I suspect.

File: index.html [hunk 10]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#splitting">Strategies for splitting up names</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p> In cultures such as parts of Southern India, Malaysia and Indonesia, a large number of people have names that consist of a given name only, with no patronym. If you require family names, you may create significant problems in these cultures, as users enter garbage data like &quot;.&quot; or &quot;Mr.&quot; in the family name field just to escape the form.</p>
```
Comment (aphillips) [inline index.html]: Add a comma after `Malaysia`

File: index.html [hunk 11]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#otherthings">Other things</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p>Some names (such as 'McNamara') contain capital letters that are not the first letter; others (such as 'van der Waals') include words that are not capitalized. Forms should preserve the case the user enters and not coerce such names to always and only use capital letters at the start of each word.</p>
```
Comment (aphillips) [inline index.html]: Consider making `always and` be `always or`

File: index.html [hunk 12]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#otherthings">Other things</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p>Allows correct capture of a family name such as that of Gabriel García Márquez, or a given name such as José María Olazábal.</p>
```
Comment (aphillips) [inline index.html]: Perhaps highlight the family and given name portions of the two exemplary names.

File: index.html [hunk 13]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#otherthings">Other things</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p>It would be wrong to assume that members of the same family share the same family name. There is a growing trend in the West for wives to keep their own name after marriage, but there are other cultures, such as China, where this is the normal approach. In some countries the wife may or may not take the husband's name.</p>
```
Comment (aphillips) [inline index.html]: Perhaps avoid considerations such as LGBT ones by genericizing "wife" and "husband", e.g.

> There is a growing trend in the West for individuals to keep their own name after marriage, but there are other cultures, such as China, where this is already the norm. In some countries a spouse may or may not choose to take the other's name (or they might do something else, such as hyphenating names)

File: index.html [hunk 14]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
+	<p class="advisement">Provide a field for a transcription of the name, where necessary.</p>
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#encoding">Implications for character support</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p>For example, Japanese users may need to provide a transcription in a Japanese syllabic script rather than/in addition to the ideographic form. This field is used for sorting Japanese names, but also allows someone looking at the name to check how it is pronounced.</p>
```
Comment (aphillips) [inline index.html:3080]: We should probably add to the sorting section... at a later date.

**Review by aphillips — CHANGES_REQUESTED** [review]
Looks great. Here's a bunch of suggestions.

File: index.html (hunk 3 above)
Comment (xfq) [inline index.html]:
```suggestion
<p>Names around the world differ greatly in composition and the order of components (see <a href="https://www.w3.org/International/questions/qa-personal-names">Personal names around the world</a>). This can create difficulties when you try to split a person's name into smaller parts for storage in a database and then retrieve them. Difficulties include understanding which part of a person's name belongs in which database field (especially when there are more or fewer parts than fields in the database), and dealing with the ordering of name parts when retrieving someone's name from the database for actual use.</p>
```

File: index.html (hunk 9 above)
Comment (xfq) [reply index.html:2931]: I think it's about the personal names in the [vCard](https://www.rfc-editor.org/info/rfc6350) and [h-card](http://microformats.org/wiki/h-card) formats.

I'm not sure what `implied "n" optimization` means here.

Comment (xfq) [reply index.html:2931]: Just found this: http://microformats.org/wiki/hcard#Implied_.22N.22_Optimization

File: index.html [hunk 15]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/International/questions/qa-personal-names#splitting">Strategies for splitting up names</a>, in <cite>Personal Names Around the World</cite>.</p>
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p>People do have names that are one letter long. These people can have problems if the form validator refuses to accept their name and demands that they supply their name in full. If you want to encourage people not to use initials, perhaps you should make that a warning message, rather than block the form submission.</p>
```
Comment (xfq) [inline index.html:2943]: Maybe it doesn't need to be written here, but I have a similar and real example.

[Pengyou](https://en.wikipedia.org/wiki/Pengyou) requires usernames to be real names. A friend of mine tried to register with his real name, but the form validator says it's not a real name and refuses to accept his name, most likely because his family name is rare. So sites shouldn't assume people's family names are in a finite set.

File: index.html [hunk 16]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
 	</details>
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
+<p>This ensures that names are correctly handled for people such as Dina Asher-Smith and Christopher O'Connell.  Note that the apostrophe may appear as <span class="codepoint" translate="no"><bdi lang="en">&#x0027;</bdi> [<span class="uname">U+0027 APOSTROPHE</span>]</span> or as <span class="codepoint" translate="no"><bdi lang="en">&#x02BC;</bdi> [<span class="uname">U+02BC MODIFIER LETTER APOSTROPHE</span>]</span>, or perhaps even <span class="codepoint" translate="no"><bdi lang="en">&#x2019;</bdi> [<span class="uname">U+2019 RIGHT SINGLE QUOTATION MARK</span>]</span>.  A hyphen may be represented using <span class="codepoint" translate="no"><bdi lang="en">&#x002D;</bdi> [<span class="uname">U+002D HYPHEN-MINUS</span>]</span> or <span class="codepoint" translate="no"><bdi lang="en">&#x2010;</bdi> [<span class="uname">U+2010 HYPHEN</span>]</span>.</p>
```
Comment (xfq) [inline index.html]: Double hyphens might also be used (in Japanese).

For foreign names that include hyphens, a double hyphen may be used instead of a hyphen. For example, if someone's family name is Zeta-Jones, when she fills in a Japanese form, her family name may appear as: `ゼタ゠ジョーンズ`.

File: index.html [hunk 17]
```diff
@@ -2825,125 +2827,265 @@ <h3>Working with personal names</h3>
...
+	</details>
+	</div>
+
+<p>The need for multiple fields will depend to some extent on what you are collecting people’s names for, and how you intend to use them.</p>
+<ul>
+<li>Are you collecting the person’s name just to have an identifier in your system?  If so, it may not matter whether the name is stored in ASCII-only or native script.</li>
+
+<li>Or will they be called by name on a welcome page or in correspondence?  If you will correspond using their name on pages written in their language, it would seem sensible to have the name in the native script.</li>
+
+<li>Is it important for people in the organization that handles queries to be able to recognise and use the person’s name?  If so, you may need a Latin transcription.</li>
```
Comment (xfq) [inline index.html]: Might be useful to make the spelling more consistent (for 'organization' and 'recognise').

File: index.html (hunk 2 above)
Comment (r12a) [reply index.html:2833]: I actually added the (s) here – they weren't in the original.  Indeed, it's to emphasise that there can be more than one of each.  However, saying "access given names" loses that message because the plurality can be applied to the idea that there are  multiple person's names. So i'm not keen on changing that.

File: index.html (hunk 3 above)
Comment (r12a) [reply index.html]: It may not involve reconstruction. A developer may try to just retrieve the family name, say for sorting.

File: index.html (hunk 2 above)
Comment (aphillips) [reply index.html:2833]: Sounds good.

File: index.html (hunk 3 above)
Comment (aphillips) [reply index.html]: Fair enough. I would probably still change this:

> This can create difficulties when you try to split a person's name into smaller parts for storage in a database and then retrieve them.

to something like:

> This can create difficulties when you try to split a person's name into smaller parts for storage and then retrieve them for use later.

(I removed "in a database" as too specific and added "for use later" just because I think it reads more clearly)

File: index.html (hunk 1 above)
Comment (r12a) [reply index.html]: Changed to:
Developers who create applications that use personal names (in web forms, databases, ontologies, and so forth) are often unaware of how different names can be in other countries.

File: index.html (hunk 3 above)
Comment (r12a) [reply index.html]: Changed to:
This can create difficulties if, for example, you try to split a person's name into smaller parts for storage in a database and then later attempt to retrieve them, especially if some reconstruction is needed.

File: index.html (hunk 4 above)
Comment (r12a) [reply index.html]: Para broken.
It links to [4.9 Defining 'string'](https://github.com/w3c/bp-i18n-specdev/pull/72#char_string) (now that i fixed the code), which i think is more directly relevant.

File: index.html (hunk 5 above)
Comment (r12a) [reply index.html]: Part of the problem is that you can't easily localise a form for a particular audience, especially for somewhere like the USA, where immigrants are arriving constantly.  You'd need to have a very high degree of personalisation to get it right.

I changed the text by dropping "in non-localized contexts" from the requirement, and rewriting the explanation as:

> Use of the terms 'first' and 'last' can be confusing for people who normally write their family name followed by given names. Although it may seem acceptable to use 'first' and 'last' for forms aimed at users in the United States, the forms may eventually be used by people with different cultural backgrounds, both within and potentially outside of the USA.

> Bear in mind, also, that in some cultures this is still problematic, such as for Icelanders, who don't actually have family names. However, short of highly localized customization, this is probably the best we can do for a generic solution.

File: index.html (hunk 6 above)
Comment (r12a) [reply index.html]: I did that, and added a link to https://www.w3.org/International/questions/qa-personal-names.en.html#patronymic

File: index.html (hunk 7 above)
Comment (r12a) [reply index.html]: Of course, when i first wrote that it was still a new kid on the block.

Replaced with: 'a social media app' (which i'm not sure was a term of art when that paragraph was first written!)

File: index.html (hunk 8 above)
Comment (r12a) [reply index.html]: Changed to:

> For example, don't assume that the order they provide names in will be 'given name' followed by 'family name', or that it will be possible in a name that is composed of multiple words to even identify which part fits into which of those categories and which parts relate to something completely different, such as a father's name, a village name, a clan name, etc.

File: index.html (hunk 9 above)
Comment (r12a) [reply index.html:2931]: Yes.

File: index.html (hunk 10 above)
Comment (r12a) [reply index.html]: Done.

File: index.html (hunk 11 above)
Comment (r12a) [reply index.html]: done

File: index.html (hunk 12 above)
Comment (r12a) [reply index.html]: done

File: index.html (hunk 13 above)
Comment (r12a) [reply index.html]: done

File: index.html (hunk 14 above)
Comment (r12a) [reply index.html:3080]: tbd

File: index.html (hunk 3 above)
Comment (r12a) [reply index.html]: done

File: index.html (hunk 15 above)
Comment (r12a) [reply index.html:2943]: I think we need a new requirement for this. Perhaps something like this:

> **Don't block unusual or unexpected names when trying to enforce real name usage.**

> It isn't hard to find examples of people who have been blocked from using a service because their name doesn't conform to expectations of the developers. If you are planning to enforce real name usage, you need to allow a mechanism for people to validate their actual names if their name is rare, or has an unexpected structure.

File: index.html (hunk 16 above)
Comment (r12a) [reply index.html]: Added.

File: index.html (hunk 17 above)
Comment (r12a) [reply index.html]: done

File: index.html (hunk 15 above)
Comment (xfq) [reply index.html:2943]: Looks good to me. Do you want to add it in this PR or a separate issue/PR?

Comment (r12a) [reply index.html:2943]: I added a commit [3] to this PR.

File: index.html [hunk 18]
```diff
@@ -2825,125 +2826,274 @@ <h3>Working with personal names</h3>
 </aside>
 
 
+<section id="loc_names_field_length">
+<h4>Field length &amp; composition</h4>
+
 	<div class="req" id="loc_names_given_family">
-	<p class="advisement">Check whether you really need to store or access given name and family name separately.</p>
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<p class="advisement">Check whether you really need to store or access given name(s) and family name(s) separately.</p>
```
Comment (aphillips) [inline index.html:2833]: I would just eliminate the parentheses and say `...access given names and family names...`

File: index.html [hunk 19]
```diff
@@ -2825,125 +2826,274 @@ <h3>Working with personal names</h3>
...
+	<details class="links"><summary>examples</summary>
+	<p><a href="https://www.w3.org/International/questions/qa-personal-names#examples">Examples of differences</a> in <cite>Personal Names Around the World</cite>.</p>
+	</details>
 	</div>
 
-	<div class="req" id="loc_names_first_last">
-	<p class="advisement">Try to avoid using the labels 'first name' and 'last name' in non-localized contexts.</p>
-	<details class="links"><summary>explanations &amp; examples</summary>
+<p>Bear in mind that names in some cultures can be quite a lot longer than your own. Make  fields long enough to enter long names. Also do not assume that a name will have more than one letter. </p>
+<p>In particular, avoid counting length in bytes (see [[[#char_string]]]) – do not assume that a four-character Japanese name in UTF-8 will fit in four bytes; you are likely to actually need 12.</p>
```
Comment (aphillips) [inline index.html:2857]: replace hyphen with colon or em dash?

Comment (r12a) [reply index.html:2857]: It's an en-dash (with the normal spacing), not a hyphen.

File: index.html (hunk 18 above)
Comment (r12a) [reply index.html:2833]: I'm trying to remind the reader that a given person's name may have more than one given and/or family name.

File: index.html (hunk 19 above)
Comment (aphillips) [reply index.html:2857]: ah... it's a British vs. American thing (we tend to use em dash where you tend to use en dashes)
